exp/api/v1beta1/types.go

```diff
@@ -122,6 +122,10 @@ type AWSLaunchTemplate struct {
 
 	// SpotMarketOptions are options for configuring AWSMachinePool instances to be run using AWS Spot instances.
 	SpotMarketOptions *infrav1.SpotMarketOptions `json:"spotMarketOptions,omitempty"`
+
+	// InstanceMetadataOptions defines the behavior for applying metadata to instances.
+	// +optional
+	InstanceMetadataOptions *infrav1.InstanceMetadataOptions `json:"metadataOptions,omitempty"`
```

Comment:
We don't need to add new fields to older API versions.  

Instead you will need to update the conversion functions so that v1beta is rountrippable to/from v1beta2.

You will need to do something similar to `AWSMachinePool`  with the [ConverrtTo](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/api/v1beta1/conversion.go#L36-L44) function and restore the value.